docs/source/reference/storage.rst

```diff
@@ -1,201 +1,131 @@
-        persistent: True  # Set to False to delete the bucket after the task is done. Defaults to True.
-        mode: MOUNT  # MOUNT or COPY. Defaults to MOUNT if not specified
+        .. code-block:: yaml
 
+          # Mount an existing S3 bucket
+          file_mounts:
+            /mydata:
+                source: s3://my-bucket/ # or gs://, r2://, cos://<region>/<bucket>
+                mode: MOUNT  # Optional - either MOUNT or COPY. Defaults to MOUNT.
 
-Common Use Cases
-^^^^^^^^^^^^^^^^
+        This will `mount <storage-mounting-modes_>`__ the contents of the bucket at ``s3://my-bucket/`` to the remote VM at ``/mydata``.
 
-1.  **You want to upload your local data to remote VM -** specify the name and
-    source fields. Name sets the bucket name that will be used, and source
-    specifies the local path to be uploaded.
 
-2.  **You want to mount an existing S3/GCS/R2 bucket to your remote VM -** specify
-    just the source field (e.g., s3://my-bucket/, gs://my-bucket/ or r2://my-bucket/).
+    .. tab-item:: Upload files to new bucket
+        :sync: new-bucket-tab
 
-3.  **You want to have a write-able path to directly write files to cloud buckets
-    -** specify a name (to create a bucket if it doesn't exist) and set the mode
-    to MOUNT. This is useful for writing code outputs, such as checkpoints or
-    logs directly to a cloud bucket.
+        To create a new bucket, upload local files to this bucket and attach it to the task,
+        specify ``name`` and ``source``, where ``source`` is a local path.
 
-4.  **You want to have a shared file-system across workers running on different
-    nodes -** specify a name (to create a bucket if it doesn't exist) and set
-    the mode to MOUNT. This will create an empty scratch space that workers
-    can write to. Any writes will show up on all worker's mount points.
+        .. code-block:: yaml
 
-Here are a few examples covering a range of use cases for sky file_mounts
-and storage mounting:
+          # Create a new S3 bucket and upload local data
+          file_mounts:
+            /mydata:
+                name: my-sky-bucket
+                source: ~/dataset   # Optional - path to local data to upload to the bucket
+                store: s3   # Optional - either of s3, gcs, r2, ibm
+                mode: MOUNT  # Optional - either MOUNT or COPY. Defaults to MOUNT.
 
-.. code-block:: yaml
+        SkyPilot will create a S3 bucket called ``my-sky-bucket`` and upload the
+        contents of ``~/dataset`` to it. The bucket will then be mounted at ``/mydata``
+        and your data will be available to the task.
 
-    name: storage-demo
+        If the bucket already exists and was created by SkyPilot, SkyPilot will fetch
+        and re-use the bucket.
 
-    resources:
-      cloud: aws
+        If ``store`` is omitted, SkyPilot will use the same cloud provider as the task's cloud.
 
+    .. tab-item:: Create empty bucket
```

Comment:
3rd tab says `Create empty bucket` and reading it, my confusion is “Ok I can do that at first sky launch; but what about subsequent launch? the bucket has been created, can I still reuse the YAML without any changes"?

Do we mean "Create or reuse" here? This is an important use case, e.g., used in https://skypilot.readthedocs.io/en/latest/examples/spot-jobs.html#an-end-to-end-example.

Reply:
good point - I've updated the title and added a visually obvious `note` box to mention if the bucket exists and was created by SkyPilot, it will be reused. Does that help?